src/unix/solarish/mod.rs

```diff
@@ -3016,24 +3016,18 @@ extern "C" {
     ) -> ::c_int;
     #[cfg_attr(
         any(target_os = "solaris", target_os = "illumos"),
-        link_name = "__posix_getpwent_r"
+        link_name = "getpwent_r"
     )]
-    pub fn getpwent_r(
-        pwd: *mut passwd,
-        buf: *mut ::c_char,
-        buflen: ::size_t,
-        result: *mut *mut passwd,
-    ) -> ::c_int;
+    pub fn native_getpwent_r(pwd: *mut passwd, buf: *mut ::c_char, buflen: ::c_int) -> *mut passwd;
```

Comment:
It's not clear to me if `native_getpwent_r()` and `native_getgrent_r()` should be `pub` or perhaps rather `pub(crate)` -- or maybe no access qualifier is necessary because they're only used from a child module?  If they're `pub` it would seem that we'd have to keep exposing them forever, because removing them would then break libc crate backwards compatibility.

Reply:
I removed the `pub` qualifier since these two functions are only used inside this child module